assets/javascripts/discourse/services/chat-draft-handler.js

```diff
@@ -0,0 +1,48 @@
+import Service from "@ember/service";
+import { ajax } from "discourse/lib/ajax";
+import { bind } from "discourse-common/utils/decorators";
+
+const DEFAULT_DRAFT = {
+  value: "",
+  uploads: [],
+  replyToMsg: null,
+};
+
+export default class ChatDraftHandler extends Service {
+  init() {
+    super.init(...arguments);
+
+    this._draftStore = {};
+    if (this.currentUser.chat_drafts) {
+      this.currentUser.chat_drafts.forEach((draft) => {
+        this._draftStore[draft.channel_id] = JSON.parse(draft.data);
+      });
+    }
+  }
+
+  @bind
+  setForChannel({ channelId, draft }) {
```

Comment:
I'm still not sure about the name here, also we say channel but we want a channelId, so it's kinda misleading.

I also think I would prefer a signature `(channelId, draft)`  same for get `(channelId)`

Reply:
I started with simple `get` and `set` but quickly realised that it overrides ember's `get` and `set` so its out of question. 

I totally agree that signature should be `(channelId, draft)` and debounce method should be in `chat-live-pane`.